frame/nomination-pools/src/lib.rs

```diff
@@ -454,56 +450,36 @@ impl<T: Config> BondedPool<T> {
 	) -> Self {
 		Self {
 			account: Self::create_account(AccountType::Bonded, depositor.clone()),
-			depositor,
-			root,
-			nominator,
-			state_toggler,
-			state: PoolState::Open,
-			points: Zero::zero(),
-			delegator_counter: Zero::zero(),
+			inner: BondedPoolInner {
+				depositor,
+				root,
+				nominator,
+				state_toggler,
+				state: PoolState::Open,
+				points: Zero::zero(),
+				delegator_counter: Zero::zero(),
+			},
 		}
 	}
 
-	// TODO: figure out if we should ditch BondedPoolStorage vs BondedPool and instead just have
-	// BondedPool that doesn't have `account` field. Instead just use deterministic accountId
-	// generation function. Only downside is this will have some increased computational cost.
-
 	/// Get [`Self`] from storage. Returns `None` if no entry for `pool_account` exists.
 	fn get(pool_account: &T::AccountId) -> Option<Self> {
-		BondedPools::<T>::try_get(pool_account).ok().map(|storage| Self {
-			points: storage.points,
-			delegator_counter: storage.delegator_counter,
-			state_toggler: storage.state_toggler,
-			depositor: storage.depositor,
-			root: storage.root,
-			nominator: storage.nominator,
-			state: storage.state,
-			account: pool_account.clone(),
-		})
+		BondedPools::<T>::try_get(pool_account)
+			.ok()
+			.map(|inner| Self { account: pool_account.clone(), inner })
```

Comment:
Just noting if we move to maps keyed by u32 Id, we can just keep the account field inn storage and ignore this whole inner thing